torchtune/modules/tokenizers/_tiktoken.py

```diff
@@ -153,5 +156,11 @@ def decode(
                 k = None
             if k:
                 token_ids = token_ids[:k]
-        token_ids = [token_id for token_id in token_ids if token_id != self.bos_id]
+        if not show_special:
```

Comment:
This is slightly far out, but could we move the logic from here to the generation recipe and move the flag for `show_special` there, too? Unless we anticipate there's other places we could use this logic, but if so, could we throw it in a utility function instead?

This way the tokenizers remain agnostic to how they are being used, and all calls to `tokenizer.decode` will behave identically.

Reply:
I see what you mean, but I do like keeping this with the tokenizer so I can quickly debug even in the middle of the recipe by calling `tokenizer.decode(sample, show_special=True)`. I find myself doing this a lot when debugging in general, and it's especially useful to see the exact text that's going into the model . If it was a separate utility it wouldn't be straightforward to do this in a pdb debugger

Reply:
Okay, yeah, IGYWM.